libnmstate/nispor/base_iface.py

```diff
@@ -0,0 +1,140 @@
+#
+# Copyright (c) 2020 Red Hat, Inc.
+#
+# This file is part of nmstate
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License as published by
+# the Free Software Foundation, either version 2.1 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with this program. If not, see <https://www.gnu.org/licenses/>.
+#
+
+import logging
+from operator import itemgetter
+
+from libnmstate.schema import Interface
+from libnmstate.schema import InterfaceIP
+from libnmstate.schema import InterfaceIPv6
+from libnmstate.schema import InterfaceState
+from libnmstate.schema import InterfaceType
+
+
+class NisporBaseIface:
+    def __init__(self, np_info):
+        self._np_info = np_info
+
+    @property
+    def np_info(self):
+        return self._np_info
+
+    @property
+    def mac(self):
+        return self._np_info.get("mac_address", "00:00:00:00:00:00")
```

Comment:
I think the better approach would be nispor python module providing objects.
We should use property of class instead of a dictionary.

Will track this in:
https://github.com/nispor/nispor/issues/10

Reply:
Nice! Thanks!

Reply:
Changed to using object property.